File: 52-hard-service-pairing/go/main.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type input struct {
	Costs [][]int64 `json:"costs"`
}

func main() {
	var in input
	if err := json.NewDecoder(os.Stdin).Decode(&in); err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
	fmt.Println(solve(in.Costs))
}

File: 52-hard-service-pairing/rust/src/main.rs
use service_pairing::solve;
use service_pairing::Input;
use std::io::{self, Read};

fn main() {
    let mut buffer = String::new();
    io::stdin().read_to_string(&mut buffer).unwrap();
    let input: Input = serde_json::from_str(&buffer).unwrap();
    println!("{}", solve(&input.costs));
}
